src/main/java/io/wispforest/owo/serialization/impl/StructCodeckBuilder.java

```diff
@@ -0,0 +1,571 @@
+package io.wispforest.owo.serialization.impl;
+
+import com.mojang.datafixers.util.*;
+import io.wispforest.owo.serialization.Codeck;
+import io.wispforest.owo.serialization.StructDeserializer;
+import io.wispforest.owo.serialization.StructSerializer;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public class StructCodeckBuilder<T> {
+
+    public static <S, F1> Codeck<S> of(StructField<S, F1> f1, Function<F1, S> constructor){
+        return new StructCodeck<S>() {
+            @Override
+            public void encode(StructSerializer struct, S value) {
+                struct.field(f1.name, f1.codec, f1.getter.apply(value));
+            }
+
+            @Override
+            public S decode(StructDeserializer struct) {
+                return constructor.apply(struct.field(f1.name, f1.codec, f1.defaultValue));
+            }
+        };
+    }
```

Comment:
Is there any way to make a struct field required?

Reply:
Fixed